simulator-starter/src/main/java/com/consol/citrus/simulator/repository/MessageRepositoryImpl.java

```diff
@@ -0,0 +1,76 @@
+package com.consol.citrus.simulator.repository;
+
+import com.consol.citrus.simulator.model.Message;
+import com.consol.citrus.simulator.model.MessageFilter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.StringUtils;
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MessageRepositoryImpl extends AbstractRepository implements MessageRepositoryCustom {
```

Comment:
hm.. we lose the JPA methods of `CrudRepository`. but you'd have to implement them when using `MessageRepository`.

Reply:
This is just the implementation of the custom part of the repository. We still have all CrudRepository functionality via the MessageRepository interface.
see https://www.baeldung.com/spring-data-composable-repositories for details

Reply:
that is cool, didn't know that. thanks for the link!